gitextensions/gitextensions
# Artificial commit changed count should be dynamic

Last planned PR for #4031 

The count for artificial commit has been hidden, only updated when the index tree was updated. The artificial count was explicitly not visible when commit count was, otherwise the count was out-of-sync. Also, the diff list of the artificial commits were not updated.

This PR reuses the setup for the count in the commit button, also refreshing the contents for artificial commits.

The PR removes an extra Task calling GetAllChangedFiles() (as the Commit button is called).
The GitStatusMonitor handling can be changed to improve performance, this is done in #4069.
There can also be improvements to caching of GetAllChangedFiles(), also a separate issue.

Note: class GitStatusMonitorUpdate kept in FormBrowse as it has knowledge of internals in FormBrowse. It could be split.

Screenshots before and after (if PR changes UI):
-
![image](https://user-images.githubusercontent.com/6248932/33796458-ef9ee210-dcf4-11e7-9d7f-6cc239c92a26.png)


How did I test this code:
 - View Artificial commits
 - View count in commit button
 - Edit a file outside of GE
 - Verify that the count updates as the count button. The revision grid is not updated instantly, there can be a second or so between the updates.

Has been tested on (remove any that don't apply):
 - Windows 10
